cli/commands/scaffold/action.go

```diff
@@ -0,0 +1,351 @@
+	moduleUrl = parsedModuleUrl.String()
+
+	// identify template url
+	templateDir := ""
+	if templateUrl != "" {
+		// process template url if was passed
+		parsedTemplateUrl, err := terraform.ToSourceUrl(templateUrl, tempDir)
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+		parsedTemplateUrl, err = rewriteTemplateUrl(opts, parsedTemplateUrl)
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+		// regenerate template url with all changes
+		templateUrl = parsedTemplateUrl.String()
+
+		templateDir, err = os.MkdirTemp("", "template")
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+		dirsToClean = append(dirsToClean, templateDir)
+
+		// downloading template
+		opts.Logger.Infof("Using template from %s", templateUrl)
+		if err := getter.GetAny(templateDir, templateUrl); err != nil {
+			return errors.WithStackTrace(err)
+		}
+	}
+
+	opts.Logger.Infof("Scaffolding a new Terragrunt module %s %s to %s", moduleUrl, templateUrl, opts.WorkingDir)
+	if err := getter.GetAny(tempDir, moduleUrl); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// extract variables from module url
+	inputs, err := config.ParseVariables(opts, tempDir)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// separate variables that require value and with default value
+	var requiredVariables []*config.ParsedVariable
+	var optionalVariables []*config.ParsedVariable
+
+	for _, value := range inputs {
+		if value.DefaultValue == "" {
+			requiredVariables = append(requiredVariables, value)
+		} else {
+			optionalVariables = append(optionalVariables, value)
+		}
+	}
+	opts.Logger.Debugf("Parsed %d required variables and %d optional variables", len(requiredVariables), len(optionalVariables))
+
+	// run boilerplate
+
+	// prepare boilerplate dir
+	boilerplateDir := util.JoinPath(tempDir, util.DefaultBoilerplateDir)
+	// use template dir as boilerplate dir
+	if templateDir != "" {
+		boilerplateDir = templateDir
+	}
+
+	// if boilerplate dir is not found, create one with default template
+	if !files.IsExistingDir(boilerplateDir) {
+		// no default boilerplate dir, create one
+		boilerplateDir, err = os.MkdirTemp("", "boilerplate")
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+		dirsToClean = append(dirsToClean, boilerplateDir)
+		if err := os.WriteFile(util.JoinPath(boilerplateDir, "terragrunt.hcl"), []byte(defaultTerragruntTemplate), 0644); err != nil {
+			return errors.WithStackTrace(err)
+		}
+		if err := os.WriteFile(util.JoinPath(boilerplateDir, "boilerplate.yml"), []byte(defaultBoilerplateConfig), 0644); err != nil {
+			return errors.WithStackTrace(err)
+		}
+	}
+
+	// add additional variables
+	vars["requiredVariables"] = requiredVariables
+	vars["optionalVariables"] = optionalVariables
+
+	vars["sourceUrl"] = moduleUrl
+
+	opts.Logger.Infof("Running boilerplate generation to %s", opts.WorkingDir)
+	boilerplateOpts := &boilerplate_options.BoilerplateOptions{
```

Comment:
IMPORTANT: please disable scripts and the `shell` helper by default. Those are a security risk and we should only turn them on if requested, via explicit flags. But those flags can be added in a separate PR later; here, we should ensure the default is safe.

Reply:
Disabled

Reply:
Sorry, I forgot one: disable hooks by default as well! Those also execute arbitrary code.

Reply:
Bumping this one more time: hooks should be disabled by default.